### PR TITLE
Add support for standardizing contest names

### PR DESCRIPTION
**Pull request by gws**
State admins would like to be able to adjust non-standard contest names uploaded via the county CVRs, so that standard contest names can be used to determine appropriate cross-county universes.

- An interface has been added to display and adjust non-standard contest names just before the contest selection page.
- When the user clicks "Save and Next", the contests are renamed in the database. This means that a user can go back to this page as many times as necessary in order to re-map contest names as new contests are uploaded.
- When the user selects the "blank" option in the drop-down, the re-mapping is not done for that contest (things are left alone).
  - Note for future improvement: if we can auto-select near-matches for the user, that would be ideal.
- If all names are standardized, the page is automatically skipped and the user sees the contest selection page.

# Notes

The React / redux style in this PR leaves **much** to be desired. For that, I'm sorry. I'm aware of many of the best practices, but a lot of the existing code / architecture lent itself to things being the way they are. If you have concrete suggestions for how to improve this that don't involve re-writing the PR, I'm all ears.

# References

- [Pivotal card](https://www.pivotaltracker.com/story/show/159142771)

### CODE REVIEW

**Comment by teaforthecat** [comment]
maybe "blank" could say "no change" or something(?)

**Comment by gws** [comment]
@teaforthecat Yeah, I think saying "-- No Change --" or something is better. I was on the fence about it but having you bring it up pushes me over the edge.

File: client/package.json [hunk 1]
```diff
@@ -47,6 +48,7 @@
         "react-addons-css-transition-group": "^15.6.0",
         "react-addons-test-utils": "^15.6.0",
         "react-dom": "^15.6.2",
+        "react-dropzone": "^4.2.13",
```
[the remark this answers is not in the document]
Comment (gws) [reply client/package.json]: Let me make sure that wasn't a bad merge. I'm tracking your branch and I didn't change `package.json` for anything in this PR.

**Comment by gws** [comment]
@sduckett Check this again. I'm basing it off of your most recent changes now - I don't believe this PR makes any `package.json` changes. If the base branch changes, then the GitHub PR looks like it includes multiple commits from your branch which make things appear to have changed for this PR when they haven't.

**Comment by gws** [comment]
I found a couple subtle issues during the recording of the demo which I resolved and I'm pretty satisfied that this is robust enough to use.

**Comment by gws** [comment]
@sduckett As I'm out for the next several days, you have all the power to merge this if and when you see fit. :+1:

**Pull request merged**